packages/amplify-e2e-tests/src/configure_tests.ts

```diff
@@ -16,12 +25,60 @@ async function setupAmplify() {
     });
     if (process.env.AWS_SESSION_TOKEN) {
       injectSessionToken('amplify-integ-test-user');
+      injectRegion('amplify-integ-test-user');
     }
   } else {
     console.log('AWS Profile is already configured');
   }
 }
 
+async function exchangeTemporaryCredentials() {
+  if (process.env.USE_PARENT_ACCOUNT) {
+    console.log('Environment is configured to use parent AWS account.');
+    return;
+  }
+
+  let accs;
+  try {
+    accs = await orgApi.listAccounts().promise();
+  } catch (e) {}
+
+  if (!accs || !accs.Accounts || accs.Accounts.length === 0) {
+    console.log('No child accounts found. Using parent AWS account.');
+    return;
+  }
+
+  const randomNumber = Math.floor(Math.random() * 100000);
+  const randomIndex = Math.floor(Math.random() * accs.Accounts.length);
+  const account = accs.Accounts[randomIndex];
+  const sts = new AWS.STS({
+    apiVersion: '2011-06-15',
+  });
+
+  // Temporary credentials will last long enough to complete a test run
+  const assumeRoleRes = await sts
+    .assumeRole({
+      RoleArn: `arn:aws:iam::${account.Id}:role/OrganizationAccountAccessRole`,
+      RoleSessionName: `testSession${randomNumber}`,
+      // One hour
+      DurationSeconds: 1 * 60 * 60,
```

Comment:
I think there may be some searchable tests where this is not long enough. I don't think there's any harm in setting it to 3 hours which is definitely enough time for all the test suites

Reply:
I like keeping the tests as short as possible. Additionally, there is an extra step required to exchange sts credentials for ones that last longer than an hour. In my testing, whenever the test lasted so long the credentials expired, I split the test into another file/step. I did this until I stopped running into issues with it expiring.